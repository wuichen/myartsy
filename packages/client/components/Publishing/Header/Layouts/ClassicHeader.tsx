import { space } from "@artsy/palette"
import { garamond } from "Assets/Fonts"
import React, { ReactElement } from "react"
import styled from "styled-components"
import { pMedia } from "../../../Helpers"
import { ClassicByline } from "../../Byline/ClassicByline"
import { ArticleData } from "../../Typings"

interface ClassicHeaderProps {
  article?: ArticleData
  date?: string
  editLeadParagraph?: ReactElement<any>
  editTitle?: ReactElement<any>
}

export const ClassicHeader: React.SFC<ClassicHeaderProps> = props => {
  const { article, date, editTitle, editLeadParagraph } = props
  return (
    <ClassicHeaderContainer>
      <Title>{editTitle || <h1>{article.title}</h1>}</Title>

      {editLeadParagraph ? (
        <LeadParagraph>{editLeadParagraph}</LeadParagraph>
      ) : (
        <LeadParagraph
          dangerouslySetInnerHTML={{ __html: article.lead_paragraph }}
        />
      )}
      <ClassicByline article={article} date={date} />
    </ClassicHeaderContainer>
  )
}

const ClassicHeaderContainer = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 900px;
  width: 100%;
  margin: ${space(4)}px auto;
  box-sizing: border-box;
  text-align: center;

  ${pMedia.xl`padding: 0 ${space(2)}px;`};

  ${pMedia.xs`
    text-align: left;
  `};
`

export const Title = styled.div`
  padding-bottom: ${space(3)}px;
  ${garamond("s37")};

  ${pMedia.xs`
    ${garamond("s34")}
  `};
`

export const LeadParagraph = styled.div`
  ${garamond("s19")};
  line-height: 1.35em;
  text-align: left;
  max-width: 580px;
  width: 100%;
  margin: 0 auto;
  padding-bottom: ${space(3)}px;

  p {
    margin: 0;
  }

  ${pMedia.xs`
    ${garamond("s17")}
    line-height: 1.35em;
  `};
`
