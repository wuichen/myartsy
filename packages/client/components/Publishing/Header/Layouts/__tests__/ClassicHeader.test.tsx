import {
  ClassicArticle,
  ClassicArticleManyAuthors,
} from "Components/Publishing/Fixtures/Articles"
import { EditableChild } from "Components/Publishing/Fixtures/Helpers"
import { mount } from "enzyme"
import "jest-styled-components"
import { cloneDeep } from "lodash"
import React from "react"
import renderer from "react-test-renderer"
import { ClassicHeader } from "../ClassicHeader"

describe("Classic Header", () => {
  describe("Snapshots", () => {
    it("renders properly", () => {
      const snapshot = renderer
        .create(<ClassicHeader article={ClassicArticle} />)
        .toJSON()
      expect(snapshot).toMatchSnapshot()
    })

    it("renders editable props properly", () => {
      const snapshot = renderer
        .create(
          <ClassicHeader
            article={ClassicArticle}
            date="2015-06-19T13:09:18.567Z"
            editLeadParagraph={EditableChild("Lead Paragraph")}
            editTitle={EditableChild("Title")}
          />
        )
        .toJSON()
      expect(snapshot).toMatchSnapshot()
    })
  })

  describe("Unit", () => {
    const getWrapper = _props => {
      return mount(<ClassicHeader {..._props} />)
    }
    let props
    beforeEach(() => {
      props = {
        article: cloneDeep(ClassicArticle),
      }
    })

    it("Renders title", () => {
      const component = getWrapper(props)
      expect(component.text()).toMatch(ClassicArticle.title)
      expect(component.text()).toMatch(ClassicArticle.author.name)
    })

    it("Renders lead paragraph", () => {
      const component = getWrapper(props)
      expect(component.text()).toMatch("Critics were skeptical of Bambi")
    })

    it("Renders author", () => {
      const component = getWrapper(props)
      expect(component.text()).toMatch(ClassicArticle.author.name)
    })

    it("Renders contributing authors", () => {
      props.article = ClassicArticleManyAuthors
      const component = getWrapper(props)
      expect(component.text()).toMatch(ClassicArticle.author.name)
      expect(component.text()).toMatch(
        "By First Author, Second Author and Third Author"
      )
    })

    describe("Editing", () => {
      it("Renders editable title", () => {
        props.editTitle = EditableChild("Title")
        const component = getWrapper(props)
        expect(component.text()).toMatch("Child Title")
      })

      it("Renders lead paragraph", () => {
        props.editLeadParagraph = EditableChild("Lead Paragraph")
        const component = getWrapper(props)
        expect(component.text()).toMatch("Child Lead Paragraph")
      })

      it("Renders a custom date", () => {
        props.date = "2017-05-19T13:09:18.567Z"
        const component = getWrapper(props)
        expect(component.text()).toMatch("May 19, 2017 9:09am")
      })
    })
  })
})
