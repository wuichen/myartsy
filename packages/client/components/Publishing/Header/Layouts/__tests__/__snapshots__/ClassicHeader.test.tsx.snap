// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classic Header Snapshots renders editable props properly 1`] = `
.c4 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: #666;
}

.c3 {
  display: block;
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 13px;
  line-height: 1.65em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 900px;
  width: 100%;
  margin: 40px auto;
  box-sizing: border-box;
  text-align: center;
}

.c1 {
  padding-bottom: 30px;
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 37px;
  line-height: 1.2em;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 19px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  line-height: 1.35em;
  text-align: left;
  max-width: 580px;
  width: 100%;
  margin: 0 auto;
  padding-bottom: 30px;
}

.c2 p {
  margin: 0;
}

@media (max-width:1280px) {
  .c0 {
    padding: 0 20px;
  }
}

@media (max-width:600px) {
  .c0 {
    text-align: left;
  }
}

@media (max-width:600px) {
  .c1 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 34px;
    line-height: 1.1em;
    -webkit-font-smoothing: antialiased;
  }
}

@media (max-width:600px) {
  .c2 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 17px;
    line-height: 1.1em;
    -webkit-font-smoothing: antialiased;
    line-height: 1.35em;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      Child 
      Title
    </div>
  </div>
  <div
    className="c2"
  >
    <div>
      Child 
      Lead Paragraph
    </div>
  </div>
  <div
    className="c3"
  >
    Joanne Artman Gallery
    <div
      className="c4"
      color="#666"
    >
      Jun 19, 2015 9:09am
    </div>
  </div>
</div>
`;

exports[`Classic Header Snapshots renders properly 1`] = `
.c4 {
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  color: #666;
}

.c3 {
  display: block;
  font-family: 'ITC Avant Garde Gothic W04','AvantGardeGothicITCW01D 731075',AvantGardeGothicITCW01Dm,Helvetica,sans-serif;
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 13px;
  line-height: 1.65em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 900px;
  width: 100%;
  margin: 40px auto;
  box-sizing: border-box;
  text-align: center;
}

.c1 {
  padding-bottom: 30px;
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 37px;
  line-height: 1.2em;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 19px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  line-height: 1.35em;
  text-align: left;
  max-width: 580px;
  width: 100%;
  margin: 0 auto;
  padding-bottom: 30px;
}

.c2 p {
  margin: 0;
}

@media (max-width:1280px) {
  .c0 {
    padding: 0 20px;
  }
}

@media (max-width:600px) {
  .c0 {
    text-align: left;
  }
}

@media (max-width:600px) {
  .c1 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 34px;
    line-height: 1.1em;
    -webkit-font-smoothing: antialiased;
  }
}

@media (max-width:600px) {
  .c2 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 17px;
    line-height: 1.1em;
    -webkit-font-smoothing: antialiased;
    line-height: 1.35em;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1>
      New Study of Yale Grads Shows the Gender Pay Gap for Artists Is Not So Simple
    </h1>
  </div>
  <div
    className="c2"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Critics were skeptical of Bambi when it was first released in 1942â€”what was the point, they wondered, of a cartoon that ignored fantasy in favor of naturalistic forest landscapes?</p>",
      }
    }
  />
  <div
    className="c3"
  >
    Joanne Artman Gallery
    <div
      className="c4"
      color="#666"
    >
      Jul 28, 2017 4:38pm
    </div>
  </div>
</div>
`;
